examples/calculations/submit_singlepoint.py

```diff
@@ -0,0 +1,168 @@
+from aiida.engine import run_get_node
+from aiida.orm import Code, Str, StructureData, load_node
+from aiida.plugins import CalculationFactory
+
+from aiida_mlip.data.model import ModelData
+
+# StructureData = DataFactory("core.structure")
+
+
+def load_model(string: Union[str, None], architecture: str) -> ModelData:
+    """
+    Load a model from a given string.
+
+    If the string represents a file path, the model will be loaded from that path.
+    Otherwise, the model will be downloaded from the specified location.
+
+    Parameters
+    ----------
+    string : Union[str, None]
+        The string representing either a file path or a URL for downloading the model.
+    architecture : str
+        The architecture of the model.
+
+    Returns
+    -------
+    ModelData or None
+        The loaded model if successful, otherwise None.
+    """
+    if string is None:
+        model = None
+    elif (file_path := Path(string)).is_file():
+        model = ModelData.local_file(file_path, architecture=architecture)
+    else:
+        model = ModelData.download(string, architecture=architecture)
+    return model
+
+
+def load_structure(value: Union[str, int, None]) -> StructureData:
+    """
+    Load a StructureData instance from the given input.
+
+    The input can be either a path to a structure file, a node PK (int),
+    or None. If the input is None, a default StructureData instance for NaCl
+    with a rocksalt structure will be created.
+
+    Parameters
+    ----------
+    value : str, int or None
+        The input value representing either a path to a structure file, a node PK,
+        or None.
+
+    Returns
+    -------
+    StructureData
+        The loaded or created StructureData instance.
+
+    Raises
+    ------
+    click.BadParameter
+        If the input is not a valid path to a structure file or a node PK.
+    """
+    if value is None:
+        structure = StructureData(ase=bulk("NaCl", "rocksalt", 5.63))
+    elif isinstance(value, int) or (isinstance(value, str) and value.isdigit()):
+        structure_pk = int(value)
+        structure = load_node(structure_pk)
+    elif Path.exists(value):
+        structure = StructureData(ase=read(value))
+    else:
+        raise click.BadParameter(
+            f"Invalid input: {value}. Must be either node PK (int) or a valid path to a structure file."
+        )
+    return structure
+
+
+def singlepoint(params: dict):
+    """
+    Prepare inputs and run a single point calculation.
+
+    Parameters
+    ----------
+    params : dict
+        A dictionary containing the input parameters for the calculations
+
+    Returns
+    -------
+    tuple
+        A tuple containing the exit code and the node of the calculation.
+    """
+    structure = load_structure(params["file"])
+
+    # Select model to use
+    model = load_model(params["model"], params["architecture"])
+
+    # Select calculation to use
+    Singlepointcalc = CalculationFactory("janus.sp")
```

Comment:
Probably ought to be camel case (`SinglepointCalc`) here and elsewhere?

Reply:
yes needs refactor SinglePointCalculation I suggest